lib/delegator.rb

```diff
@@ -0,0 +1,18 @@
+class Delegator
+    def initialize(parsers)
+        # @parsers = args.fetch(:parsers)
+        @parsers = parsers
+    end
+
+    def delegate(input, game)
+        if contains_parser(input, game)
+            parser = @parsers[game.state.to_sym]
+            parser.parse(input, game)
+        end
+    end
+
+    private
+    def contains_parser(input, game)
+        return @parsers.keys.include? game.state
```

Comment:
- New line after private
- Don't need the return
- if the function is like a boolean one or a 'question' add a question mark at the end `contains_parser?`